File: NTTU_ISMS_OJ/test/create_cpp_file.cpp
#include <iostream>
#include <string>
#include <fstream>
using namespace std;
int main() {
    string str, filename, index;
    cout << "Input Filename (without Filename extension): ";
    getline(cin, str);
    filename = (str + ".cpp");
    cout << "File name = " << filename << endl;

    ofstream outFile(filename, ios::out);
    if(!outFile) {cout << "file open error" << '\n';} 

    else {
        cout << "Input sorce code :\n";
        do {
            getline(cin, index);
            outFile << index << '\n' << flush;
        } while (index != "}");
    }

    outFile.close();
    system("PAUSE");
    return 0;
}

File: NTTU_ISMS_OJ/test/create_cpp_file.py
#coding=UTF-8
import os
from Extension_modules import file_directory

filename = str(input("Input Filename (without Filename extension): ") + ".cpp")
print("File name = " + filename)
filename = file_directory.path_function("/{}".format(filename))
print("File path = " + filename)

outFile = open(filename, 'w')
index = str(input("Input sorce code : \n"))
outFile.write("{}\n".format(index))
outFile.flush()

outFile = open(filename, 'a')
while (index != '}'):
    index = str(input())
    outFile.write("{}\n".format(index))
    outFile.flush()

outFile.close()
os.system("PAUSE")